src/bundle/Controller/ContentTypeController.php

```diff
@@ -385,6 +391,63 @@ public function editAction(Request $request, ContentTypeGroup $group, ContentTyp
         ]);
     }
 
+    /**
+     * @param \eZ\Publish\API\Repository\Values\ContentType\ContentTypeGroup $group
+     * @param \eZ\Publish\API\Repository\Values\ContentType\ContentType $contentType
+     *
+     * @return \Symfony\Component\HttpFoundation\Response
+     *
+     * @throws \eZ\Publish\API\Repository\Exceptions\UnauthorizedException
+     */
+    public function copyAction(Request $request, ContentTypeGroup $group, ContentType $contentType): Response
+    {
+        $this->denyAccessUnlessGranted(new Attribute('class', 'create'));
+
+        $contentTypeService = $this->contentTypeService;
+        $notificationHandler = $this->notificationHandler;
+
+        $copyData = new ContentTypeCopyData($contentType, $group);
+
+        $form = $this->contentTypeFormFactory->contentTypeCopy($copyData);
+        $form->handleRequest($request);
+
+        if ($form->isSubmitted()) {
+            $result = $this->submitHandler->handle($form, function (ContentTypeCopyData $data) use ($contentTypeService, $notificationHandler) {
+                $contentType = $data->getContentType();
+
+                try {
+                    $contentTypeService->copyContentType($contentType);
+
+                    $notificationHandler->success(
+                        /** @Desc("Content Type '%name%' copied.") */
+                        'content_type.copy.success',
+                        ['%name%' => $contentType->getName()],
+                        'content_type'
+                    );
+                } catch (UnauthorizedException $exception) {
```

Comment:
When this exception can occur? Is `$this->denyAccessUnlessGranted(new Attribute('class', 'create'));` not enough?